### PR TITLE
refactor isHostResourceType to not receive the context from reconciler and not leak types

### PR DESCRIPTION
**Pull request by gnoff**
type validateDOMNesting
move `isHostResourceType` to ReactDOMHostConfig
type `AncestorInfo`
refactor `resourceFormOnly` into `ancestorInfo.containerTagInScope`
provide hostContext from reconciler

### CODE REVIEW

**Review by sebmarkbage — APPROVED** [review]
nobody:

sizebot: +0.01%

File: packages/react-dom-bindings/src/client/validateDOMNesting.js [hunk 1]
```diff
@@ -155,20 +179,23 @@ if (__DEV__) {
...
+    containerTagInScope: null,
   };
 
-  updatedAncestorInfo = function(oldInfo, tag) {
-    const ancestorInfo = {...(oldInfo || emptyAncestorInfo)};
+  updatedAncestorInfoDev = function(
+    oldInfo: ?AncestorInfoDevType,
+    tag: string,
+  ) {
+    const AncestorInfoDev = {...(oldInfo || emptyAncestorInfoDev)};
```
Comment (sebmarkbage) [inline packages/react-dom-bindings/src/client/validateDOMNesting.js:189]: Why is this capitalized. Seems like an odd naming convention?

Comment (gnoff) [reply packages/react-dom-bindings/src/client/validateDOMNesting.js:189]: I thought so too. I think it's meant to function like a singleton kind of like how Dispatcher is capitalized. It made my type name worse though so I changed it but then changed it back...

It's not a singleton but in memory layout but how it is accessed it's sort of a single global value so it made sense in that (possibly wildly inaccurate) mental framing and I restored it

Comment (gnoff) [reply packages/react-dom-bindings/src/client/validateDOMNesting.js:189]: waitaminute....

Comment (gnoff) [reply packages/react-dom-bindings/src/client/validateDOMNesting.js:189]: I did a find replace and capitalized it all myself and then convinced myself it was always that way. brb...

Comment (gnoff) [reply packages/react-dom-bindings/src/client/validateDOMNesting.js]: 🤦

**Pull request merged**